fdbclient/NativeAPI.actor.cpp

```diff
@@ -1321,7 +1321,29 @@ Reference<ProxyInfo> DatabaseContext::getMasterProxies(bool useProvisionalProxie
 			masterProxies = Reference<ProxyInfo>( new ProxyInfo( clientInfo->get().proxies, false ) );
 			grvProxies = Reference<ProxyInfo>( new ProxyInfo( clientInfo->get().proxies, true ) );
 			provisional = clientInfo->get().proxies[0].provisional;
+
+			// if any TSS present, update model's endpoint mapping
+			std::unordered_map<uint64_t, Endpoint> tssEndpointMap;
+			if (!clientInfo->get().tssMapping.empty()) {
```

Comment:
you also need a check for when a new storage server interface is loaded in `StorageServerInfo::getInterface`